### PR TITLE
Prebuild Python, just like Ruby

### PR DESCRIPTION
**Pull request by kylekthompson**
Follows the work done in https://github.com/rwx-research/mint-leaves/pull/47, but does it for Python in addition to Ruby.

### CODE REVIEW

File: .mint/python-binary-builder.yml [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+      set -ueo pipefail
+
+      # install dependencies
+      sudo apt-get update
+      sudo apt-get install \
+        ca-certificates \
+        dpkg-dev \
+        gcc \
+        gnupg \
+        libbluetooth-dev \
```
Comment (ayazhafiz) [inline .mint/python-binary-builder.yml:24]: 🤯

File: .mint/python-binary-builder.yml [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+        "EXTRA_CFLAGS=${EXTRA_CFLAGS:-}" \
+        "LDFLAGS=${LDFLAGS:-}" \
+        "PROFILE_TASK=${PROFILE_TASK:-}"
+
+      sudo rm python
+      sudo make -j "$nproc" \
+        "EXTRA_CFLAGS=${EXTRA_CFLAGS:-}" \
+        "LDFLAGS=${LDFLAGS:--Wl},-rpath='\$\$ORIGIN/../lib'" \
+        "PROFILE_TASK=${PROFILE_TASK:-}" \
+        python
```
Comment (ayazhafiz) [inline .mint/python-binary-builder.yml:92]: Just curious, are the two makes necessary? Is the second `make ... python` insufficient?

Comment (kylekthompson) [reply .mint/python-binary-builder.yml:92]: I'm not sure. I think this was ripped from a Python dockerfile and I'm terrified to change it 😅

Comment (kylekthompson) [reply .mint/python-binary-builder.yml:92]: I'd like to _think_ they probably do different things

**Review by ayazhafiz — APPROVED** [review]

File: .mint/python-binary-builder.yml (hunk 2 above)
Comment (kylekthompson) [reply .mint/python-binary-builder.yml:92]: https://github.com/docker-library/python/blob/7fe6c4b20ab3173a08299df64987af410bdc8923/3.12/bookworm/Dockerfile#L54-L72

**Pull request merged**